odc/stac/eo3/_eo3converter.py

```diff
@@ -210,24 +212,29 @@ def _to_dataset(
     if crs is None:
         crs = EPSG4326
 
+    if asset_url is None:
```

Comment:
Would it be worth accounting for the possibility of a pystac Item already having an `href`? Or do we know that will never be the case?

Reply:
that's a good point, I didn't try if either stac item got `href` in it, or loading could work with a `uris` pointing to `odc-yaml`.  Had it been, then there is no need to get url from stac path. I'll try it now.